doc/PostgreSQL_backend.txt

```diff
@@ -0,0 +1,129 @@
+flat files.
+
+Todo
+----
+- Adapt rollback to work with PostgreSQL, since rollback normally used SQLite3
+- Adapt minetestmapper to use PostgreSQL
+- Test under Windows
+- Adapt mods to use PostgreSQL rather than flat files, when possible
+
+How to help
+-----------
+Please email if this document could be improved in any way or any possible ideas
+that could help.
+
+Who might benefit
+-----------------
+Anyone who is running a dedicated long term server with a regular number of
+players on at all times. Typically this will be a Linux server.
+
+Advantages of PostgreSQL backend:
+---------------------------------
+- Greater use of memory to improve performance
+- Automatic reindexing using the auto vacuum feature
+- Backups while minetest users are playing
+- Continous backups, using automatic transaction log archiving
+- Ability use seperate application and database server, for increased performance
+- Redundancy, load balancing, and failover, via PostgreSQL features
+- Use of triggers and additional conliums or tables
+
+Further documentation
+----------------------
+- PostgreSQL: http://www.postgresql.org/
+- Continous Archiving: http://www.postgresql.org/docs/9.4/static/continuous-archiving.html
+- Tuning: https://wiki.postgresql.org/wiki/Tuning_Your_PostgreSQL_Server
+- Kernel Resources: http://www.postgresql.org/docs/9.4/static/kernel-resources.html
+- Auto Vacuum: http://www.postgresql.org/docs/9.4/static/runtime-config-autovacuum.html
+- High Availability, Load Balancing, and Replication: http://www.postgresql.org/docs/9.4/static/high-availability.html
+
+How to compile
+--------------
+
+This document is not going to be extensive enough to explain the details of
+compiling open source projects. There should be plenty of information on the
+web that explains that much better.
+
+I will explain how to compile Minetest with the PostgreSQL backend on Ubuntu
+14.10. These settings are minimal and meant to be combined with your normal
+biuld configuration.
+
+# apt-get install cmake cmake-data libpq-dev postgresql-client postgresql-server-dev
+
+# cmake -DBUILD_SERVER=TRUE -DBUILD_CLIENT=FALSE -DUSE_SOUND=FALSE -DENABLE_POSTGRESQL=TRUE
+
+# make
+
+Configuration
+-------------
+
+The settings for PostgreSQL user, password, database name, and database host
+are located in the world.mt file. These are the settings.
+
+backend =	postgresql
+pg_host =	database hostname, defaults to socket connection
+pg_port =	port number, defaults to 5432
+pg_dbname  =	minetest database name, defults to minetest
+pg_username = 	user name if different from the user running minetestserver,
+		defaults to minetest
+pg_password =	password, if md5 password required (see pg_hba.conf)
+
+Schema
+------
+
+This is all that is required.
+
+CREATE TABLE IF NOT EXISTS blocks (
+	pos BIGINT PRIMARY KEY,
+	data BYTEA
+);
+
+On my server I save the timestamp for when block data was originally generated
```

Comment:
You should perhaps put a Headline "Additional note" over this. Its a nice gimmick therefore it can be kept, but it should be more separate from "official" documentation.